platform/default/local_file_request.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <mbgl/storage/file_source_request.hpp>
+#include <mbgl/storage/response.hpp>
+#include <mbgl/util/string.hpp>
+#include <mbgl/util/io.hpp>
+#include <mbgl/util/url.hpp>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#if defined(_WINDOWS) && !defined(S_ISDIR)
+#define S_ISDIR(m) (((m) & S_IFMT) == S_IFDIR)
+#endif
+
+namespace mbgl {
+
+static void requestLocalFile(const std::string& path, Response& response, ActorRef<FileSourceRequest> req) {
+    struct stat buf;
+    int result = stat(path.c_str(), &buf);
+
+    if (result == 0 && (S_IFDIR & buf.st_mode)) {
+        response.error = std::make_unique<Response::Error>(Response::Error::Reason::NotFound);
+    } else if (result == -1 && errno == ENOENT) {
+        response.error = std::make_unique<Response::Error>(Response::Error::Reason::NotFound);
+    } else {
+        auto data = util::readFile(path);
+        if (!data) {
+            response.error = std::make_unique<Response::Error>(
+                Response::Error::Reason::Other,
+                std::string("Cannot read file ") + path);
+        } else {
+            response.data = std::make_shared<std::string>(std::move(*data));
+        }
+    }
+
+    req.invoke(&FileSourceRequest::setResponse, response);
+}
```

Comment:
Why is this implementation in a header file?

Reply:
moved to designated cpp file